ping.go

```diff
@@ -492,18 +476,32 @@ func (p *Pinger) processPacket(recv *packet) error {
 
 	switch pkt := m.Body.(type) {
 	case *icmp.Echo:
-		data := IcmpData{}
-		err := json.Unmarshal(m.Body.(*icmp.Echo).Data, &data)
-		if err != nil {
-			return err
+
+		// If we are priviledged, we can match icmp.ID
+		if p.network == "ip" {
+			// Check if reply from same ID
+			if pkt.ID != p.id {
+				return nil
+			}
+		}
+
+		if len(pkt.Data) < timeSliceLength+trackerLength {
+			return errors.New("insufficient data received")
```

Comment:
could you change this to `fmt.Errorf` to avoid adding in the errors package?

Reply:
please also add some details about the packet that might be useful for debugging

Reply:
For future reference, are you concerned about the binary size or just a larger `import` section?
As an aside, technically `errors` is already included by [`fmt`](https://golang.org/src/fmt/print.go) so there is no binary growth by importing it.

Reply:
ah, I didn't realize fmt was already using errors, thx for the heads-up